Fix/header widgets optimization

Tim-M-S opened the pull request:
### All changes meet the following requirements
- [x] Changelog entry was added
- [x] Changes have been tested
- [x] Plugin can be built

@plentymarkets/ceres-io

Tim-M-S committed:
optimized header widgets behavior both in sb and frontend

Tim-M-S committed:
minimal css change

Tim-M-S committed:
header scrolling is now debounced

Tim-M-S committed:
improved usability of widget-fw class

Tim-M-S committed:
added firefox compatibility

felixdausch reviewed (commented):
![Bildschirmfoto 2019-10-17 um 15 45 37](https://user-images.githubusercontent.com/7404578/67014470-32b6b800-f0f5-11e9-9bab-c6b45a40068b.png)
- [x] Grid mit .widget-fw => Abstand am Rand
- [x] Grid > Widget mit .widget-fw => seltsames verhalten. Evtl. .widget-fw auf oberste Ebene beschränken.

felixdausch reviewed (changes requested):
s.o.

Tim-M-S committed:
removed firefox specific code
widget-fw only styles first level widgets
fixed padding issues
added check if scrollbar is visible

Tim-M-S committed:
removed code

felixdausch commented on resources/scss/ceres/widgets/Helper/_base-widget.scss:
```diff
@@ -51,26 +51,48 @@
-{
+#page-header-parent > .widget-fw,
+#page-body > .container-max > .row > .col > .widget-fw,
+.footer > .row > .col > .widget-fw {
```
durch diese sehr strikten Selektor-Ketten ist die Wahrscheinlichkeit sehr groß, dass durch Anpassungen in Themes die Utility-Klasse nicht mehr funktioniert. Das sollte nicht der Fall sein. Ich würde hier wie vorher nur die Klasse `.widget-fw` selektieren und über `.widget .widget-fw` die Formatierungen wieder aufheben, um zu verhindern, dass die Klasse auf unteren ebenen verwendet wird.

Tim-M-S replied:
done

felixdausch commented on resources/scss/ceres/widgets/Helper/_base-widget.scss:
```diff
@@ -51,26 +51,48 @@
-{
-    margin-bottom: 0;
+
+html.scrollbar .widget-fw {
```
Die Klasse`.scrollbar` sorgt dafür, dass das raster nicht mehr korrekt ausgerichtet ist:

**html.scrollbar**
![Bildschirmfoto 2019-10-23 um 08 27 11](https://user-images.githubusercontent.com/7404578/67364021-01177400-f56f-11e9-9d39-e5e5254dbac1.png)

**html:not(.scrollbar)**
![Bildschirmfoto 2019-10-23 um 08 27 04](https://user-images.githubusercontent.com/7404578/67364061-11c7ea00-f56f-11e9-8299-0d45e81a566a.png)

Tim-M-S replied:
Lässt sich nicht nachstellen

felixdausch commented on resources/views/Widgets/Common/SeparatorWidget.twig:
```diff
@@ -5,5 +5,5 @@
 
 {% if isPreview %}<div class="widget-editable-min-height">{% endif %}
-    <hr{% if inlineSpacings | length > 0 %} style="{{ inlineSpacings }}"{% endif %}{% if spacingClasses | length > 0 %} class="{{ spacingClasses }}{% if customClass | length > 0 %} {{ customClass }}{% endif %}"{% endif %}>
+    <hr class="widget widget-separator{% if spacingClasses | length > 0 %} {{ spacingClasses }}{% endif %}{% if customClass | length > 0 %} {{ customClass }}{% endif %}" {% if inlineSpacings | length > 0 %} style="{{ inlineSpacings }}"{% endif %}>
```
Die Klasse `.widget` setzt ein margin-bottom von 2rem. Dadurch verändert sich das Standardverhalten der Trennlinie. Ist das gewollt/ gefordert?

Tim-M-S replied:
done

Tim-M-S committed:
changes requested by f.dausch

Tim-M-S committed:
removed check for scrollbar
